src/cpdb/cpdb.py

```diff
@@ -1,17 +1,156 @@
 import streamlit as st  # type: ignore
 import pandas as pd
-from src.cpdb.helpers import get_data
+from src.cpdb.helpers import get_data, get_commit_cols, get_diff_dataframe, get_map_percent_diff, viz_keys
+import plotly.express as px
+import plotly.graph_objects as go
+
 
 def cpdb():
     st.title("Capital Projects DB QAQC")
     branch = st.sidebar.selectbox(
         "select a branch",
         ['main']
     )
-    data = get_data(branch)
+    agency_label = {'sagency': "Sponsoring Agency", 'magency': "Managing Agency"}
+    agency_type = st.sidebar.selectbox(
+        "select an agency type",
+        ['sagency', 'magency'],
+        format_func=lambda x: agency_label.get(x)
+    )
+    view_type = st.sidebar.selectbox(
+        "select to view by projects or commitment",
+        ['projects', 'commitments']
+    )
+    
+    subcategory = st.sidebar.selectbox(
+        "choose a subcategoy or entire ",
+        ['all categories', 'fixed assets']
+    )
+    
+    data = get_data(branch, agency_type)
+    
+    st.caption(
+        body="""There are mainly three ways to look at the existing qaqc tables. 
+        First, you can either focus on the managing agencies or the sponsoring agencies for the projects. 
+        Second, you can choose either to focus on the number of projects by your choice of agency type or the commitment amount for the projects. 
+        Third, you could view the only a subcategory of the projects that outlined by the Capital Planning database. 
+        """
+    )
+    st.header(f'Sorted By Number of {view_type} in the Latest {agency_type} Summary Stats')
+
+    df = data[agency_type].set_index(agency_type + "acro")
+    df_pre = data["pre_" + agency_type].set_index(agency_type + "acro")
+    if view_type == "commitments":
+        df = df[get_commit_cols(df)]
+        df_pre = df_pre[get_commit_cols(df_pre)]
+    else: 
+        df.drop(labels=get_commit_cols(df), axis=1, inplace=True)
+        df_pre.drop(labels=get_commit_cols(df_pre), axis=1, inplace=True)
+
+    # sort the values based on projects/commitments and get the top ten agencies
+    df_bar = df.sort_values(by=viz_keys[subcategory][view_type]["values"][0], ascending=False)
```

Comment:
agree on this. My thinking in the long term is this could be tied to larger refactoring i.e. with a more object-oriented approach and a class method like sorting can probably do this better. But for now i will just create a standalone function.